src/core/contracts/identity.test.ts

```diff
@@ -166,6 +171,119 @@ describe("identity", () => {
       expect(await isAuthorizedTo(NON_OWNER, contractAddress, 1, 60)).toBeTruthy();
     });
   });
+
+  describe("#getDomainSeperator", () => {
+    it("returns a EIP712 domain seperator", async () => {
+      const fakeContractAddress = "0xfake";
+      const expected = {
+        chainId: expect.any(Number),
+        name: "Identity",
+        salt: "0xa0bec69846cdcc8c1ba1eb93be1c5728385a9e26062a73e238b1beda189ac4c9",
+        verifyingContract: "0xfake",
+        version: "1",
+      };
+
+      expect(await getDomainSeperator(fakeContractAddress)).toEqual(expected);
+    });
+  });
+
+  describe("#createAddDelegateEip712TypedData", () => {
+    let contractAddress: EthereumAddress;
+    let contractOwner: EthereumAddress;
+
+    beforeAll(async () => {
+      contractOwner = await signer.getAddress();
+      const identityContract = await new Identity__factory(signer).deploy(contractOwner);
+      await identityContract.deployed();
+      contractAddress = identityContract.address;
+    });
+
+    it("creates an EIP712 data type", async () => {
+      const message = {
+        delegateAddr: "0x3c44cdddb6a900fa2b585dd299e03d12fa4293bc",
+        nonce: 1,
+        role: 0x1,
+      };
+
+      const expected = {
+        types: {
+          EIP712Domain: [
+            { name: "name", type: "string" },
+            { name: "version", type: "string" },
+            { name: "chainId", type: "uint256" },
+            { name: "verifyingContract", type: "address" },
+            { name: "salt", type: "bytes32" },
+          ],
+          DelegateAdd: [
+            { name: "nonce", type: "uint32" },
+            { name: "delegateAddr", type: "address" },
+            { name: "role", type: "uint8" },
+          ],
+        },
+        primaryType: "DelegateAdd",
+        domain: {
+          chainId: expect.any(Number),
+          name: "Identity",
+          salt: "0xa0bec69846cdcc8c1ba1eb93be1c5728385a9e26062a73e238b1beda189ac4c9",
+          verifyingContract: contractAddress,
+          version: "1",
+        },
+        message: {
+          delegateAddr: "0x3c44cdddb6a900fa2b585dd299e03d12fa4293bc",
+          nonce: 1,
+          role: 0x1,
+        },
+      };
+
+      expect(await createAddDelegateEip712TypedData(contractAddress, message)).toEqual(expected);
+    });
+  });
```

Comment:
Since the DelegateAddParams lets you provide your own nonce, I'd like to see a test of #createAddDelegateEip712TypedData that supplies a nonce in the message and shows that it's used.